### ISSUE TITLE
Scaffolding fails: "The annotation 'MySql:IndexPrefixLength' cannot be added because an annotation with the same name already exists."

### ISSUE THREAD

**Issue by engstrand** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1189]
Scaffolding existing DB that was successfully scaffolded using MS package in same project:

`Scaffold-DbContext "server=localhost;user=user;password=password;database=mysql_db" "Pomelo.EntityFrameworkCore.MySql" -OutputDir EF`

The process ends pretty much immiediately with exception and stacktrace below.

**Exception message**
 System.InvalidOperationException: The annotation 'MySql:IndexPrefixLength' cannot be added because an annotation with the same name already exists.

**Stack trace**
```
 at Microsoft.EntityFrameworkCore.Infrastructure.Annotatable.AddAnnotation(String name, Annotation annotation)
   at Microsoft.EntityFrameworkCore.Infrastructure.Annotatable.AddAnnotation(String name, Object value)
   at Microsoft.EntityFrameworkCore.Infrastructure.Annotatable.Microsoft.EntityFrameworkCore.Metadata.IMutableAnnotatable.AddAnnotation(String name, Object value)
   at Microsoft.EntityFrameworkCore.MutableAnnotatableExtensions.AddAnnotations(IMutableAnnotatable annotatable, IEnumerable`1 annotations)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.RelationalScaffoldingModelFactory.VisitIndex(EntityTypeBuilder builder, DatabaseIndex index)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.RelationalScaffoldingModelFactory.VisitIndexes(EntityTypeBuilder builder, ICollection`1 indexes)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.RelationalScaffoldingModelFactory.VisitTable(ModelBuilder modelBuilder, DatabaseTable table)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.RelationalScaffoldingModelFactory.VisitTables(ModelBuilder modelBuilder, ICollection`1 tables)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.RelationalScaffoldingModelFactory.VisitDatabaseModel(ModelBuilder modelBuilder, DatabaseModel databaseModel)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.RelationalScaffoldingModelFactory.Create(DatabaseModel databaseModel, Boolean useDatabaseNames)
   at Microsoft.EntityFrameworkCore.Scaffolding.Internal.ReverseEngineerScaffolder.ScaffoldModel(String connectionString, DatabaseModelFactoryOptions databaseOptions, ModelReverseEngineerOptions modelOptions, ModelCodeGenerationOptions codeOptions)
   at Microsoft.EntityFrameworkCore.Design.Internal.DatabaseOperations.ScaffoldContext(String provider, String connectionString, String outputDir, String outputContextDir, String dbContextClassName, IEnumerable`1 schemas, IEnumerable`1 tables, Boolean useDataAnnotations, Boolean overwriteFiles, Boolean useDatabaseNames)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.ScaffoldContextImpl(String provider, String connectionString, String outputDir, String outputDbContextDir, String dbContextClassName, IEnumerable`1 schemaFilters, IEnumerable`1 tableFilters, Boolean useDataAnnotations, Boolean overwriteFiles, Boolean useDatabaseNames)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.ScaffoldContext.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
```

### Further technical details

MySQL version: 5.7.18
Operating system: Windows 10
Pomelo.EntityFrameworkCore.MySql version: 3.2.2
Microsoft.AspNetCore.App version: 3.1

**Comment by mguinness** [comment]
The interesting thing is that `Pomelo.EntityFrameworkCore.MySql` isn't even shown in stack trace,  What happens if you create a new project and try to scaffold the db?

**Comment by engstrand** [comment]
I have created a new project but the result was the same.

**Comment by engstrand** [comment]
Could it be related to this: https://github.com/dotnet/efcore/issues/11846

**Comment by mguinness** [comment]
Can you isolate it to a certain table w/ the [Tables](https://docs.microsoft.com/en-us/ef/core/miscellaneous/cli/powershell#scaffold-dbcontext) option?  If so, please share the schema of the table.

**Comment by engstrand** [comment]
Great, that is useful information! 
I could check my tables and maybe modify/remove some index to get around this.
But how come this went well using MS packages?

**Comment by lauxjpn** [comment]
> But how come this went well using MS packages?

I don't think SQL Server uses index prefixes at all. I guess you must have converted the MSSQL database somehow to MySQL. In the conversion process, the index prefixes must have been added (which should be correct).

--

The fix will make it into the `3.2.3` version which will be released tomorrow.
